src/Parser/Shaman/Enhancement/Modules/Items/SmolderingHeart.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import ITEMS from 'common/ITEMS';
+import SPELLS from 'common/SPELLS';
+import ItemIcon from 'common/ItemIcon';
+import ItemLink from 'common/ItemLink';
+import { formatPercentage } from 'common/format';
+
+const SMOLDERING =
+  {PROC_CHANCE : 0.0012,
+    PROC_DURATION : 10000};
+
+class SmolderingHeart extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  smolderingProcs = 0;
+  smolderingDuration = 0;
+  maelstromSpent = 0;
+  ascendanceCastEventIds = new Set();
+  furyMaelstromUsed = 0;
+  lastStormstrikeTimestamp;
+  lastWindstrikeTimestamp;
+
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasHands(ITEMS.SMOLDERING_HEART.id);
+  }
+
+  smoldering_uptime() {
+    return this.smolderingDuration / this.owner.fightDuration;
+  }
+
+  expected_smoldering_uptime() {
+    // Smoldering Heart (Enhancement) gives a 0.12% (0.0012 decimal) chance per point of Maelstrom spent to grant Ascendance for 10 seconds (10000ms).
+    const EXPECTED_PROC_COUNT = (this.maelstromSpent)*SMOLDERING.PROC_CHANCE;
+    const EXPECTED_UPTIME = (EXPECTED_PROC_COUNT * SMOLDERING.PROC_DURATION) / this.owner.fightDuration;
+    return EXPECTED_UPTIME;
+  }
+
+  on_byPlayer_cast(event) {
+      if (event.ability.guid === SPELLS.ASCENDANCE_TALENT_ENHANCEMENT.id) {
+          this.ascendanceCastEventIds.add(event.eventUniqueId);
+      }
+      if (event.ability.guid === SPELLS.STORMSTRIKE.id) {
+          this.maelstromSpent += 40;
+      } else if (event.ability.guid === SPELLS.LAVA_LASH.id) {
+          this.maelstromSpent += 30;
+      } else if (event.ability.guid === SPELLS.CRASH_LIGHTNING.id || event.ability.guid === SPELLS.EARTHEN_SPIKE_TALENT.id || event.ability.guid === SPELLS.FROSTBRAND.id || event.ability.guid === SPELLS.SUNDERING_TALENT.id) {
+          this.maelstromSpent += 20;
+      } else if (event.ability.guid === SPELLS.WINDSTRIKE.id) {
+          this.maelstromSpent += 8; //8 due to 80% cost reduction in Ascendance
+      }
+  }
+
+  on_byPlayer_applybuff(event) {
+    if (event.ability.guid === SPELLS.ASCENDANCE_TALENT_ENHANCEMENT.id) {
```

Comment:
I am not 100% on this but cant you look for ascendance applyBuffs that didnt follow a ascendance cast?